Makefile

```diff
@@ -99,7 +108,7 @@ proxy_handler_test:
 clean:
 	rm -rf *.o *.a *~ *.gch *.swp *.dSYM *.gcda *.gcno *.gcov config_parser config_parser_test *.tar.gz webserver request_handler.o session.o request_handler_test webserver_test session_test echo_handler_test file_handler_test response_test request_test not_found_handler_test session_test http_client_test proxy_handler_test
 
-test: clean default
+test: clean echo_handler_test file_handler_test not_found_handler_test response_test request_test session_test http_client_test proxy_handler_test
```

Comment:
It's good that we separate out the tests. Might make it faster to compile with the default make.